### DISCUSSION TITLE
Invoke an edge function before sending OTP

### DISCUSSION THREAD

**Discussion by danesposi-pandor** [supabase/supabase#21243] [Questions]
Hey everyone!. I'm trying to call an edge function that tests the validity of the email an user provides when requesting an OTP. This validations consist of allowing only certain email domains into the app, checking the MX records of the domain, checking a custom blacklist, etc. 

I want to send OTPs only if the user passes these validations. I read you can make HTTP calls from within supabase functions, and these functions can be later used in triggers, but I'm not sure if this is considered best practice.

Thank you in advance!.

**Comment by GaryAustin1** [comment]
You can have an insert trigger function on auth.users (similar to the user management one many use for populating a profile table).
In that function you can do all kinds of checks and raise an exception to abort the signup before the OTP is sent.
If you need an edge function then you can call that with the http extension.  The only issue is last time I checked the auth process has a 2 or 3 second timeout on the original database call or it aborts the operation.

**Reply by danesposi-pandor** [reply]
Got it. So let's say I'd like to look into an external database for a certain email domain before sending an OTP, this would require an external call. Knowing that the auth process has this timeout, how would you recommend to proceed here?

**Reply by GaryAustin1** [reply]
You would use the http extension in your trigger function to call that API and get the result back.  Then raise an exception if bad.

This is a very old discussion on it https://github.com/orgs/supabase/discussions/4143
There are others here.  And the http extensions is documented.
